### PR TITLE
RUN-1680: prevent cross-origin errors (+ RUN-1682)

### PR DESCRIPTION
**Pull request by Domiii**
* Fix for: https://linear.app/replay/issue/RUN-1680#comment-ac407129
* Added `IsInReplayCode` API function to replace `RecordReplayAllowAccesses` (https://linear.app/replay/issue/RUN-1502/new-runtime-api-isinreplaycode)
* Sneaky changes:
  * RUN-1682 (one line): https://linear.app/replay/issue/RUN-1682/js-exception-in-pages-with-iframes
  * Added `recordreplay::Warning` (because I used it temporarily)
* Tests look good ✔

### Previous tests
* Tests: 100+ tests passed. Only `weather.com` failed. But that is beign flakey in general. ✔

### CODE REVIEW

**Review by bhackett1024 — CHANGES_REQUESTED** [review]
This seems like the wrong fix, and basically rolls back @klochek 's work to surface these JS exceptions as crashes.  Can we try fixing this within V8 so the exception doesn't get thrown, and clients don't get incorrect stacks?

**Comment by Domiii** [comment]
@bhackett1024 
> This seems like the wrong fix

I would not call it a "wrong fix". Its a "let's quickly get rid of these crashes" type of fix, which is much in line with our previous strategy. I am not at all opposed to spending more time on this and make it better, and I will!

However, I don't agree with the word "wrong" 😊

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 1]
```diff
@@ -1275,7 +1288,7 @@ function previewBlinkNode(node) {
   }
 
   let childNodes;
-  if (node.nodeName == "IFRAME") {
+  if (node.nodeName == "IFRAME" && node.contentDocument) {
```
Comment (Domiii) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:1291]: I stumbled across this by accident. This should fix `RUN-1682`.

File: base/record_replay.h [hunk 2]
```diff
@@ -97,7 +98,7 @@ void OnNavigationEvent(const char* kind, const char* url);
 int NewIdMainThread(const char* name);
 int NewIdAnyThread(const char* name);
 
-
+bool AllowCrossDomainAccesses();
```
Comment (bhackett1024) [inline base/record_replay.h:101]: If we're making this part of the recordreplay API let's use the IsInReplayCode() name from RUN-1502, this name is too specific to the bindings code.

**Review by bhackett1024 — APPROVED** [review]

**Pull request merged**